tmt/steps/finish/shell.py

```diff
@@ -0,0 +1,63 @@
+import fmf
+import tmt
+import click
+
+
+class FinishShell(tmt.steps.finish.FinishPlugin):
+    """
+    Perform finishing tasks using shell scripts
+
+    Example config:
+
+    finish:
+        how: shell
+        script:
+            - upload-logs.sh
+            - rm -rf /tmp/temporary-files
+
+    Use the 'order' attribute to select in which order finishing tasks
+    should happen if there are multiple configs. Default order is '50'.
+    """
+
+    # Supported methods
+    _methods = [tmt.steps.Method(name='shell', doc=__doc__, order=50)]
+
+    @classmethod
+    def options(cls, how=None):
+        """ Finish command line options """
+        return [
+            click.option(
+                '-s', '--script', metavar='SCRIPT',
```

Comment:
To me this looks like a "prepare" phase  (or inverse thereof) again. But the usecase might not be necessarily there. So I'm not against leaving just a "scriptlet" allowed for finish.

Reply:
We should support scripts as well, but let's add that as a separate pull request.